### PR TITLE
Device clean: remove onApiResponse()

### PR DESCRIPTION
**Pull request by AntonPuko**

### CODE REVIEW

File: src/server/DeviceServer_v2.js [hunk 1]
```diff
@@ -299,21 +299,21 @@ class DeviceServer {
...
-      );
-
-      if (global.api) {
-        global.api.safeMode(
-          deviceID,
-          systemMessage.getPayload().toString(),
-        );
-      }
-    }
+    // if (lowerEventName.indexOf('spark/device/safemode') === 0) {
```
Comment (AntonPuko) [inline src/server/DeviceServer_v2.js:302]: commented this until figuring out global.api softCoap things

File: src/clients/Device.js [hunk 2]
```diff
@@ -723,39 +541,68 @@ class Device extends EventEmitter {
     return Messages.getResponseType(request);
   };
 
+  // todo make return type annotation
+  getDescription = async (): Promise<*> => {
+    const isBusy = !this._isSocketAvailable(null);
+    if (isBusy) {
+      throw new Error('This core is locked during the flashing process.');
```
Comment (AntonPuko) [inline src/clients/Device.js:548]: put this in the most of newly added public functions...
i could wrap it into function like `_checkIsDeviceBusy()` - but it seems less clean

File: src/clients/Device.js [hunk 3]
```diff
@@ -723,39 +541,68 @@ class Device extends EventEmitter {
...
       null,
       messageToken,
     );
+
     return this._transformVariableResult(name, message);
   };
 
-  _setVariable = async (
+  // TODO refactor, make sure if we need this at all
+  setVariableValue = async (
```
Comment (AntonPuko) [inline src/clients/Device.js:596]: not sure if we need this at all...now we use it `nowhere` in the codebase..and I'm not sure if this works at all. Generally, if it's completely written it'd be different from `getVariableValue` only by putting payload as 3rd arg here:
const token = this.sendMessage('VariableRequest', { name }, payload);

File: src/clients/Device.js [hunk 4]
```diff
@@ -766,55 +613,69 @@ class Device extends EventEmitter {
...
 
   /**
    * Asks the core to start or stop its 'raise your hand' signal
-   * @param showSignal - whether it should show the signal or not
-   * @param callback - what to call when we're done or timed out...
    */
-  _raiseYourHand = async (showSignal: boolean, ..._:void[]): Promise<*> => {
+    // TODO figure out what is it for and why messageName isn't
+    // consistent with messages specification
+  raiseYourHand = async (
```
Comment (AntonPuko) [inline src/clients/Device.js:670]: what's the purpose of it? we don't use it anywhere and more over `_raiseYourHand` messageName doesn't fit with messages specification.

File: src/clients/Device.js (hunk 3 above)
Comment (jlkalberer) [reply src/clients/Device.js:596]: Yeah, keep for now

File: src/clients/Device.js (hunk 4 above)
Comment (jlkalberer) [reply src/clients/Device.js:670]: Yeah, I'm guessing this has to do with handshake.. or something on the device to see if the server is alive.

**Review by jlkalberer — APPROVED** [review]

File: src/server/DeviceServer_v2.js (hunk 1 above)
Comment (durielz) [reply src/server/DeviceServer_v2.js:302]: that code and the other one regarding the spark/device/claim/code event were introduced by me in the old client/SparkCore.js. global.api.safemode and global.api.linkDevice refer to 2 methods in views/api_v1.js in spark-server i implemented in order to manage the safemode status of a device and the claim process. also i publish the spark/device/system/version event from my devices every time they start in order to save the current firmware system version

File: src/clients/Device.js (hunk 4 above)
Comment (durielz) [reply src/clients/Device.js:670]: sending the raiseYourHand message generate a SIGNAL_START or SIGNAL_STOP coap message like the Signal button in Particle IDE triggering a rainbow LED pattern on your device.
if you fix your code in this way it works:
`_raiseYourHand = async (showSignal: boolean, ..._:void[]): Promise<*> => {
    const token = this.sendMessage(
      'RaiseYourHand',
      { _writeCoapUri: Messages.raiseYourHandUrlGenerator(showSignal) },
      null,
    );
    return await this.listenFor(
      'RaiseYourHandReturn',
      null,
      token,
    );
  };`

Comment (AntonPuko) [reply src/clients/Device.js:670]: Thanks for the explanation.
Tried..nasty rainbow :)...so, its just like `visual` pinging the device?

Comment (durielz) [reply src/clients/Device.js:670]: Yes! I added the signal param (0|1) to the PUT endpoint /devices/DEVICE_ID in order to use it with the API. I think Particle did the same but i haven't tried yet.

File: src/server/DeviceServer_v2.js (hunk 1 above)
Comment (jlkalberer) [reply src/server/DeviceServer_v2.js:302]: Ahh... I started working on the claim process from the cli but there are bugs with the windows implementation where Windows 10 will not connect to the device sometimes.

I'd love to get the safemode and version working. We aren't saving the correct version right now in our implementation.

File: src/clients/Device.js (hunk 4 above)
Comment (jlkalberer) [reply src/clients/Device.js:670]: Thanks @durielz - I think we'll just ask you when we have questions haha. You seem to know everything about the CLI/Firmware

Comment (durielz) [reply src/clients/Device.js:670]: ahah, yes i worked a lot on the server and protocol, i made a product based on photon. but i like the way you are refactoring the code.

**Pull request merged**